src/awkward/contents/unionarray.py

```diff
@@ -243,16 +249,7 @@ def simplified(
                                     contents[k].length,
                                 )
                             )
-                            old_parameters = contents[k]._parameters
-                            contents[k] = (
-                                contents[k]
-                                ._mergemany([inner_cont])
-                                .copy(
-                                    parameters=ak._util.merge_parameters(
-                                        old_parameters, inner_cont._parameters
-                                    )
-                                )
-                            )
+                            contents[k] = contents[k]._mergemany([inner_cont])
```

Comment:
I've removed the parameter merging logic here. I don't fully understand _why_ we did things differently before, so I'd appreciate a sense check here @jpivarski

Reply:
I wrote the above code with parameter-merging to make the `simplified` classmethod do what the old `simplify` method used to do. (It was a translation, not a new design.)

`UnionArray.simplify_union` last existed in 1157d228629ff481b7ae26c9653e45655ef11420, and the above line came from

https://github.com/scikit-hep/awkward/blob/1157d228629ff481b7ae26c9653e45655ef11420/src/awkward/contents/unionarray.py#L533

So no, there was no specialized logic here about merging parameters. I was probably thinking "this has to be done because we don't want to lose parameters" but I had forgotten that `_mergemany` merges parameters internally, so I unnecessarily did it another time.

Therefore, it's safe to remove here.